### DISCUSSION TITLE
folders moved in remote git repo don't get synced using wiki.js Force Sync operation

### DISCUSSION THREAD

**Discussion by iamtennislover** [requarks/wiki#3279] [Error / Bug Report]
**Describe the bug**
I have my wiki.js setup with bi-directional remote git repo (using bitbucket). If folders were moved from the git repo separately outside the wiki.js local repo (using `git mv`), the wiki.js's UI sync operation is not moving the folders and files in the postgres when I click "Force Sync"

**To Reproduce**
1. Have your wiki.js setup with git setup:
  - auth type: ssh
  - sync direction: bi-directional
2. Ensure both remote git repo and wiki.js's local git repo are in sync by going to wiki.js home page (from Chrome browser) Account -> Admin -> Storage -> Clicking "Force Sync"
3. Confirm the local and remote repo are in sync by using `git log` if needed
4. Create a test folder and test files in remote repo
```
➜  my-separate-repo git:(master) ✗ git pull
➜  my-separate-repo git:(master) ✗ mkdir -p foobar/test1
➜  my-separate-repo git:(master) ✗ echo foo1 > foobar/test1/test2.md
➜  my-separate-repo git:(master) ✗ echo foo2 > foobar/test1/test3.md
➜  my-separate-repo git:(master) ✗ git add foobar
➜  my-separate-repo git:(master) ✗ git commit -m 'uploading foobar folder to remote repo - separate from wiki.js repo' -a
...
➜  my-separate-repo git:(master) ✗ git push
...
```
5. Pull remote repo changes to wiki.js from home page (from Chrome browser) Account -> Admin -> Storage -> Clicking "Force Sync"
6. Confirm the new files and new folders are created by navigating to http://localhost:8000/en/foobar/test1/test2
7. Now, from remote repo, delete the entire folder
```
➜  my-separate-repo git:(master) ✗ git pull
➜  my-separate-repo git:(master) ✗ git mv foobar foobar99
➜  my-separate-repo git:(master) ✗ ls foobar99/
test1
➜  my-separate-repo git:(master) ✗ git commit -m 'renaming foobar - separate from wiki.js repo' -a
➜  my-separate-repo git:(master) ✗ git push
```
8. Pull remote repo changes to wiki.js from home page (from Chrome browser) Account -> Admin -> Storage -> Clicking "Force Sync"
10. You can see that `foobar99` doesn't get created and also stale `foobar` exists in the wiki.js, where as in git remote repo `foobar` no longer exists and only `foobar99` exists
9. Here are the logs when this pull happens: (***NOTE THE WARNING MESSAGES: `Failed to process`***)
```
2020-12-12T19:29:53.654Z [MASTER] info: (STORAGE/GIT) Performing pull rebase from origin on branch master...
2020-12-12T19:29:55.829Z [MASTER] info: (STORAGE/GIT) Performing push to origin on branch master...
2020-12-12T19:29:57.213Z [MASTER] warn: (STORAGE/GIT) Failed to process {foobar => foobar99}/test1/test2.md
2020-12-12T19:29:57.213Z [MASTER] warn: ENOENT: no such file or directory, open '/Users/booha/wikijs/data/repo/{foobar => foobar99}/test1/test2.md'
2020-12-12T19:29:57.214Z [MASTER] warn: (STORAGE/GIT) Failed to process {foobar => foobar99}/test1/test3.md
2020-12-12T19:29:57.214Z [MASTER] warn: ENOENT: no such file or directory, open '/Users/booha/wikijs/data/repo/{foobar => foobar99}/test1/test3.md'
```
10. Any further sync operations lead to no warning messages going forward:
```
2020-12-12T19:33:04.483Z [MASTER] info: (STORAGE/GIT) Performing pull rebase from origin on branch master...
2020-12-12T19:33:06.956Z [MASTER] info: (STORAGE/GIT) Performing push to origin on branch master...
2020-12-12T19:33:07.481Z [MASTER] info: Syncing with storage target git...
2020-12-12T19:33:07.504Z [MASTER] info: (STORAGE/GIT) Performing pull rebase from origin on branch master...
2020-12-12T19:33:09.320Z [MASTER] info: (STORAGE/GIT) Performing push to origin on branch master...
2020-12-12T19:33:11.125Z [MASTER] info: Syncing with storage target git: [ COMPLETED ]
```
11. Finally, confirm from the wiki.js local git repo, git is in sync, so wiki.js seems to think that all is good, but in reality, postsql is in out of sync of git repo (as confirmed the wiki.js home webpage)
```
➜  wikijs cd /Users/booha/wikijs/data/repo
➜  repo git:(master) git
➜  repo git:(master) git log --oneline -n 10
3f908d2 (HEAD -> master, origin/master) renaming foobar - separate from wiki.js repo
0fa57bf uploading foobar folder to remote repo - separate from wiki.js repo
...
```
12. wiki.js home page shows `foobar` folder (but in reality, `foobar99` should be seen)

**Expected behavior**
regardless of what type of git operation is performed in git remote repo, if git remote repo should be considered as authoritative source and clicking "Force Sync" from wiki.js should pull down the latest data from git remote repo and push it to its postgres DB

**Screenshots**
Not required

**Host Info (please complete the following information):**
 - OS: Mac OS - Darwin (darwin) 19.6.0 x64
 - Wiki.js version: Wiki.js 2.5.170
 - Database engine: postgres 12.3 localhost
 - Node.js 14.5.0

**Additional context**
Whats worst is, I cannot bulk delete an entire folder, so as a workaround, I have to delete each file under the old `foobar` folder manually from the wiki.js to update the postgres DB
1. Also When deleting the files under `foobar` from wiki.js page http://localhost:8000/en/foobar/test1/test2, I get this error: `fatal: pathspec './foobar/test1/test2.md' did not match any files` on the wepage, but atleast postgres DB gets updated
2. I also delete http://localhost:8000/en/foobar/test1/test3 with above error message,
3. finally `foobar` is not shown in the wiki.js webpage

**Comment by iamtennislover** [comment]
one point to call out is that this only happens with `git mv`, and with `git rm` this issue doesn't happen

**Comment by zoenglinghou** [comment]
I have the same issue. If you are running a small-scale wiki, maybe you could try resetting the database?

**Comment by peteruithoven** [comment]
Duplicate of https://github.com/Requarks/wiki/discussions/3707 ?
